src/braket/circuits/gates.py

```diff
@@ -59,6 +59,63 @@
 
 # Single qubit gates #
 
+class Barrier(Gate):
+    r"""Barrier gate.
+    """
+    def __init__(self, qubit_count):
+        super().__init__(qubit_count=qubit_count, ascii_symbols=[f"||" for i in range(qubit_count)])
+
+    def bind_values(self, **kwargs):
+        raise NotImplementedError
+
+    @property
+    def _qasm_name(self):
+        return f"barrier"
+
+    def __hash__(self):
+        return hash((self.name, self.qubit_count, self.qubit_count))
+
+    @staticmethod
+    @circuit.subroutine(register=True)
+    def barrier(target):
+        r"""Barrier gate.
+        Args:
+            target: Target qubit(s)
+        """
+        return Instruction(Barrier(len(target)), target=target)
+    
+
+Gate.register_gate(Barrier)
+
+class Delay(Gate):
+    r"""Delay gate. Applies delay in ns.
```

Comment:
We should specify `delay` as a `float` in seconds, to be consistent with the existing pulse API for `delay`: https://github.com/amazon-braket/amazon-braket-sdk-python/blob/1c46ca7f18f68b963ba06edf7bb719258e838a16/src/braket/pulse/pulse_sequence.py#L166-L181

Reply:
Yes, when serializing a duration to OpenQASM, the unit must be included (i.e., 0.01 --> `0.01s`)